### PR TITLE
Add customer to RevolutPayOptions for data prefill

### PR DESCRIPTION
**Pull request by andrewbents**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -446,6 +454,8 @@ export interface RevolutPayOptions extends CommonOptions {
   email?: string
   /** Styles of the RevolutPay button */
   buttonStyle?: ButtonStyleOptions
+  /** Prefiiled person details of user. Only valid details are prefilled. */
+  prefilledPersonalDetails?: PrefilledPersonalDetails
```
Comment (ohansemmanuel) [inline src/types.ts:458]: can we please rename this to `customer` ? It'll be inline with other widget method e.g., card popup.

Comment (ohansemmanuel) [reply src/types.ts:458]: e.g., as close to the existing `CustomerDetails` type here (line 367) as possible

File: src/types.ts [hunk 2]
```diff
@@ -446,6 +454,8 @@ export interface RevolutPayOptions extends CommonOptions {
   email?: string
   /** Styles of the RevolutPay button */
   buttonStyle?: ButtonStyleOptions
+  /** Prefiiled customer details. Only valid details are prefilled. */
+  customer?: PrefilledCustomerDetails
```
Comment (ohansemmanuel) [inline src/types.ts:458]: why not: 

```ts 
customer?: CustomerDetails & {dateOfBirth: string}
```
with a tsdoc comment on the string format type for the dob? as with other existing customer details properties? 
CleanShot 2022-11-08 at 10 58 50

Comment (ohansemmanuel) [reply src/types.ts:458]: Or better still add `dateOfBirth` to the existing `CustomerDetails` type - with a comment stating its used only for revpay?

**These will keep our exposed APIs as consistent as possible**

Comment (andrewbents) [reply src/types.ts:458]: `CustomerDetails` has `name`. How would we get `firstName` and `lastName` in this case? I'm not keen to using `.split(' ')`

File: src/types.ts [hunk 3]
```diff
@@ -371,6 +371,8 @@ export interface CustomerDetails {
   email?: string
   /** Customer's phone number if available */
   phone?: string
+  /** Customer's date of birth. Currently used only in RevolutPay. */
```
Comment (ohansemmanuel) [inline src/types.ts:374]:
```suggestion
  /** Customer's date of birth in the format DD/MM/YY. Currently used only in Revolut Pay. */
```

**Review by ohansemmanuel — APPROVED** [review]

**Pull request merged**